Mathlib/Probability/Kernel/Disintegration/CdfToKernel.lean

```diff
@@ -0,0 +1,690 @@
+
+lemma lintegral_stieltjesOfMeasurableRat [IsFiniteKernel κ] (hf : IsRatCondKernelCDF f κ ν)
+    (a : α) (x : ℝ) :
+    ∫⁻ b, ENNReal.ofReal (stieltjesOfMeasurableRat f hf.measurable (a, b) x) ∂(ν a)
+      = κ a (univ ×ˢ Iic x) := by
+  rw [← set_lintegral_univ, set_lintegral_stieltjesOfMeasurableRat hf _ _ MeasurableSet.univ]
+
+lemma integrable_stieltjesOfMeasurableRat [IsFiniteKernel κ] (hf : IsRatCondKernelCDF f κ ν)
+    (a : α) (x : ℝ) :
+    Integrable (fun b ↦ stieltjesOfMeasurableRat f hf.measurable (a, b) x) (ν a) := by
+  have : (fun b ↦ stieltjesOfMeasurableRat f hf.measurable (a, b) x)
+      = fun b ↦ (ENNReal.ofReal (stieltjesOfMeasurableRat f hf.measurable (a, b) x)).toReal := by
+    ext t
+    rw [ENNReal.toReal_ofReal]
+    exact stieltjesOfMeasurableRat_nonneg _ _ _
+  rw [this]
+  refine integrable_toReal_of_lintegral_ne_top ?_ ?_
+  · refine (Measurable.ennreal_ofReal ?_).aemeasurable
+    exact (measurable_stieltjesOfMeasurableRat hf.measurable x).comp measurable_prod_mk_left
+  · rw [lintegral_stieltjesOfMeasurableRat hf]
+    exact measure_ne_top _ _
+
+lemma set_integral_stieltjesOfMeasurableRat [IsFiniteKernel κ] (hf : IsRatCondKernelCDF f κ ν)
+    (a : α) (x : ℝ) {s : Set β} (hs : MeasurableSet s) :
+    ∫ b in s, stieltjesOfMeasurableRat f hf.measurable (a, b) x ∂(ν a)
+      = (κ a (s ×ˢ Iic x)).toReal := by
+  rw [← ENNReal.ofReal_eq_ofReal_iff, ENNReal.ofReal_toReal]
+  rotate_left
+  · exact measure_ne_top _ _
+  · exact set_integral_nonneg hs (fun _ _ ↦ stieltjesOfMeasurableRat_nonneg _ _ _)
+  · exact ENNReal.toReal_nonneg
+  rw [ofReal_integral_eq_lintegral_ofReal, set_lintegral_stieltjesOfMeasurableRat hf _ _ hs]
+  · exact (integrable_stieltjesOfMeasurableRat hf _ _).restrict
+  · exact ae_of_all _ (fun _ ↦ stieltjesOfMeasurableRat_nonneg _ _ _)
+
+lemma integral_stieltjesOfMeasurableRat [IsFiniteKernel κ] (hf : IsRatCondKernelCDF f κ ν)
+    (a : α) (x : ℝ) :
+    ∫ b, stieltjesOfMeasurableRat f hf.measurable (a, b) x ∂(ν a)
+      = (κ a (univ ×ˢ Iic x)).toReal := by
+  rw [← integral_univ, set_integral_stieltjesOfMeasurableRat hf _ _ MeasurableSet.univ]
+
+end stieltjesOfMeasurableRat
+
+section isRatCondKernelCDFAux
+
+variable {f : α × β → ℚ → ℝ}
+
+/-- This property implies `IsRatCondKernelCDF`. The measurability, integrability and integral
+conditions are the same, but the limit properties of `IsRatCondKernelCDF` are replaced by
+limits of integrals. -/
+structure isRatCondKernelCDFAux (f : α × β → ℚ → ℝ) (κ : kernel α (β × ℝ)) (ν : kernel α β) :
+    Prop :=
+  (measurable : Measurable f)
+  (mono' (a : α) {q r : ℚ} (_hqr : q ≤ r) : ∀ᵐ c ∂(ν a), f (a, c) q ≤ f (a, c) r)
+  (nonneg' (a : α) (q : ℚ) : ∀ᵐ c ∂(ν a), 0 ≤ f (a, c) q)
+  (le_one' (a : α) (q : ℚ) : ∀ᵐ c ∂(ν a), f (a, c) q ≤ 1)
+  (tendsto_integral_of_antitone (a : α) (s : ℕ → ℚ) (_hs : Antitone s)
+    (_hs_tendsto : Tendsto s atTop atBot) :
+    Tendsto (fun m ↦ ∫ c, f (a, c) (s m) ∂(ν a)) atTop (𝓝 0))
```

Comment:
Isn't this assumption the same as requiring that the integral `∫ c, f (a, c) q ∂(ν a)` tends to `0` when `q` tends to -\infty? As it's just an auxiliary construction, I agree that you should just take the condition which is the most convenient for the proofs, but it might be worth pointing this out in some comment.

Also, if you stick with sequences, could you avoid the letter `s` for the sequence, as it's a set in the rest of the file?

Reply:
I left comments to explain why the assumption is written this way and renamed the variables.